plugin/Gui/Table/EstateListTable.php

```diff
@@ -226,9 +226,10 @@ public function get_columns()
 	 *
 	 */
 
-	protected function column_shortcode($pItem)
+	protected function column_shortcode( $pItem )
 	{
-		return '<input type="text" readonly value="[oo_estate view=&quot;'.esc_html($pItem->name).'&quot;]">';
+		return '<input type="text" style="max-width: 100%; margin-right: 5px;" readonly value="[oo_estate view=&quot;' . esc_html( $pItem->name ) . '&quot;]"><input type="button" class="button button-copy" data-clipboard-text="[oo_estate view=&quot;' . esc_html( $pItem->name ) . '&quot;]" value="' . esc_html__( 'Copy',
+				'onoffice-for-wp-websites' ) . '" >';
```

Comment:
I just noticed that the button becomes visible slightly after the rest of the page has loaded. I didn't think of that. But there seems to be a solution: If you use an inline script right after the button and set the class there, it will happen before the first paint. So could you move the check whether the button should be shown from the script file into an internal script?

For example, you could change this line to:
```
return '<input type="text" style="max-width: 100%; margin-right: 5px;" readonly value="[oo_estate view=&quot;' . esc_html( $pItem->name ) . '&quot;]"><input type="button" class="button button-copy" data-clipboard-text="[oo_estate view=&quot;' . esc_html( $pItem->name ) . '&quot;]" value="' . esc_html__( 'Copy',
				'onoffice-for-wp-websites' ) . '" >
<script>if (navigator.clipboard) { jQuery('.button-copy').show(); }';</script>
```

You would need to make this change for all places where the button is rendered. You can then remove this check from the script (https://github.com/onOfficeGmbH/oo-wp-plugin/blob/44f9412b1c06f016dd9373564c90e8e9762658dc/js/onoffice-copycode.js).

Does this make sense?

Reply:
Yes I will update this